prestoadmin/server.py

```diff
@@ -219,34 +219,51 @@ def check_presto_version():
     Returns:
         Error string if applicable
     """
-    version = get_presto_version()
-    if version in PRESTO_TD_RPM:
-        return ''
-    try:
-        # remove -SNAPSHOT or .SNAPSHOT from the version string
-        version = re.sub(r'[-\.]SNAPSHOT.*', '', version)
-        float(version)
-        version_number = version.strip().split('.')
-        if int(version_number[1]) < PRESTO_RPM_MIN_REQUIRED_VERSION:
-            incorrect_version_str = 'Presto version is %s, version >= 0.%d ' \
-                                    'required.' % \
-                                    (version, PRESTO_RPM_MIN_REQUIRED_VERSION)
-            warn(incorrect_version_str)
-            return incorrect_version_str
-        return ''
-    except ValueError:
+    if not presto_installed():
         not_installed_str = 'Presto is not installed.'
         warn(not_installed_str)
         return not_installed_str
 
+    version = get_presto_version()
+    if version in PRESTO_TD_RPM:
+        return ''
+
+    matched = re.search('(\d+)\.(\d+)t?([-\.]SNAPSHOT)?', version)
+    if not matched:
+        incorrect_version_str = 'Incorrect presto version:  %s,' % version
+        warn(incorrect_version_str)
+        return incorrect_version_str
+
+    major_version = matched.group(1)
+    minor_version = matched.group(2)
+
+    assert major_version.isdigit(), 'incorrect major presto version'
```

Comment:
actually it should never happens, because of the reg-exp. It is just some kind of paranoia-check :-) if you don't like it i would better remove it at all =)